cmd/mesh/manifest-migrate.go

```diff
@@ -15,23 +15,131 @@
 package mesh
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 
+	"github.com/ghodss/yaml"
 	"github.com/spf13/cobra"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/kubectlcmd"
+	"istio.io/operator/pkg/translate"
+	"istio.io/operator/pkg/util"
+	"istio.io/operator/pkg/version"
+)
+
+const (
+	defaultNamespace = "istio-system"
 )
 
-func manifestMigrateCmd(rootArgs *rootArgs) *cobra.Command {
+type manifestMigrateArgs struct {
+	// migrateDir indicates whether to migrate all values.yaml files in directory
+	migrateDir bool
```

Comment:
You could eliminate this flag and test if the given path is a dir.

Reply:
actually I think specifying this flag makes it more clear to user that what they are expecting. But it depends on whether user use it more often with directory or single file. If they use it more often with directory then we should eliminate this flag.

Reply:
consider a situation like the flag is set but the path is a file - you have to document it and do additional error handling. imo it's better to document the behavior to avoid such inconsistencies and keep the api such that it's not possible to create inconsistent parameter sets, but it's up to you.